packages/@lwc/engine-dom/src/renderer.ts

```diff
@@ -142,12 +142,11 @@ function setProperty(node: Node, key: string, value: any): void {
     if (process.env.NODE_ENV !== 'production') {
         if (node instanceof Element && !(key in node)) {
             // TODO [#1297]: Move this validation to the compiler
-            assert.fail(
-                `Unknown public property "${key}" of element <${
-                    node.tagName
-                }>. This is likely a typo on the corresponding attribute "${htmlPropertyToAttribute(
+            // eslint-disable-next-line no-console
+            console.warn(
+                `Unknown public property "${key}" of element <${node.tagName.toLowerCase()}>. This is either a typo on the corresponding attribute "${htmlPropertyToAttribute(
                     key
-                )}".`
+                )}", or the attribute does not exist in this browser or DOM implementation.`
```

Comment:
I would love to use a generic logging utility here, but that lives in `@lwc/engine-core`, not `@lwc/engine-dom`. In `@lwc/engine-dom`, we seem to have a mix of `console.warn`s and `console.errors`, with `eslint-disable-next-line no-console` added for each.

We should really clean those up, but I chose to punt on that for this PR.

We can also debate whether this should be `console.warn` or `console.error`, but I like `warn` here because it's more clear that this is just a helpful warning message, not something going drastically wrong.

Reply:
This validation is moving to `@lwc/engine-core` in https://github.com/salesforce/lwc/pull/2997. I will update it to use `logWarn` if your PR gets merged first.